### PR TITLE
API server deployment for RDMA test cases

### PR DESCRIPTION
**Pull request by thomas-am**
### Description of PR

Summary:
Add support to deploy (and remove) a Keysight IxNetwork API Server docker container.
This is required for the RDMA series of test cases.

### Type of change

- [ ] Bug fix
- [x] Testbed and Framework(new/improvement)
- [ ] Test case(new/improvement)

#### What is the motivation for this PR?

Prior to running RDMA test cases, the Keysight API server needs to be deployed.

#### How did you do it?

Deployment was integrated into the existing testbed-cli add/remove-topo workflow 
that used for PTF containers already.

#### How did you verify/test it?

Tested deployment and removal of API Server, including download of 
the "docker-keysight-api-server" image from the docker registry.

Also verified that existing:
 a) add/remove-topo with PTF containers, and
 b) start/stop-vms
still works as before.

#### Any platform specific information?

No.

#### Supported testbed topology if it's a new test case?

N/A

### Documentation 

Yes, see update in ansible/doc/README.testbed.Cli.md

### CODE REVIEW

File: ansible/roles/vm_set/tasks/main.yml [hunk 1]
```diff
@@ -156,53 +156,68 @@
...
+    when: VM_num is defined and VM_num|int > 0
+
+  - name: Generate vm list of target VMs
+    set_fact: VM_targets={{ VM_hosts | filter_vm_targets(topology['VMs'], VM_base) | sort }}
+    when: topology['VMs'] is defined
+
+  when: vm_required is defined and vm_required == True
+
 
 - name: Stop VMs
```
Comment (neethajohn) [inline ansible/roles/vm_set/tasks/main.yml:222]: We need to add checks to ensure that we don't run into any weird behavior if the user tries to specify any of the subcommands - stop, start etc when the ptf image is apiserver

File: ansible/doc/README.testbed.Cli.md [hunk 2]
```diff
@@ -48,3 +48,21 @@ Goal is to use one VM set against different DUTs
 Feature: The VMs configuration will NOT be updated while switching from one topo to another (faster).
 
 TODO: check topo field when renumbering between topologies
+
+# Deploy Ixia IxNetwork API server
```
Comment (neethajohn) [inline ansible/doc/README.testbed.Cli.md:52]: Can you explicity call out that this topo will not have any VMs?

File: ansible/roles/vm_set/tasks/main.yml [hunk 3]
```diff
@@ -237,15 +246,9 @@
...
-- name: Connect VMs
-  include_tasks: connect_vms.yml
-  when: action == 'connect_vms'
-
-- name: Disconnect VMs
-  include_tasks: disconnect_vms.yml
-  when: action == 'disconnect_vms'
+  when:
+    - action == 'renumber_topo'
+    - container_type is defined and container_type == "PTF"
```
Comment (neethajohn) [inline ansible/roles/vm_set/tasks/main.yml:251]: can move this also inside the 'vm_required == True' block and get rid of the 'container_type' var

Comment (thomas-am) [reply ansible/roles/vm_set/tasks/main.yml:251]: let me do that...

File: ansible/roles/vm_set/tasks/main.yml [hunk 4]
```diff
@@ -156,61 +156,91 @@
...
-  include_tasks: stop.yml
-  when: action == 'stop'
-
-- name: Start VMs
-  include_tasks: start.yml
-  when: action == 'start'
+- name: Require VMs by default
+  set_fact:
+    vm_required: true
+    container_type: "PTF"
```
Comment (neethajohn) [inline ansible/roles/vm_set/tasks/main.yml:162]: This var seems to be redundant. Can we get rid of this?

Comment (thomas-am) [reply ansible/roles/vm_set/tasks/main.yml]: done

File: ansible/roles/vm_set/tasks/main.yml [hunk 5]
```diff
@@ -156,61 +156,89 @@
...
+
+    - name: Disconnect VMs
+      include_tasks: disconnect_vms.yml
+      when: action == 'disconnect_vms'
+
+    - name: Renumber topology
+      include_tasks: renumber_topo.yml
+      when:
+        - action == 'renumber_topo'
+        - container_type is defined and container_type == "PTF"
```
Comment (neethajohn) [inline ansible/roles/vm_set/tasks/main.yml:239]: please remove this one too

Comment (thomas-am) [reply ansible/roles/vm_set/tasks/main.yml]: done - thanks for noticing

**Review by neethajohn — APPROVED** [review]

**Pull request merged**